ls1intum/Artemis
# [Bugfix] Fix counts within Exam Checklist and Locking bugs with programming exercises

### Checklist

### Motivation and Context
A few bugs were discovered recently and we aim to fix them in this PR.

1. Currently test runs are still included in the counts, which should not be the case. We removed them by using the correct query

2. A index out of bounds exception: When the user tried to access a result of the first correction round, when the result of the second correction round did already existed aswell this exception was thrown due to an erroneous check. `>=` was needed instead of `==`.

3. When the only result (a manual one) of a programmingSubmission was cancelled, the submission was deleted as well. At some point in the second correction development, a new submission was created for each new manual assessment. Since this is not the case anymore we mustn't delete the submission anymore.


### Description


### Steps for Testing


1. Make sure that within the exam checklist no counts (especially within the assessment row) include the test runs.
2. go to an exam, to a programming exercise -> exercise-groups -> submissions. In this dashboard click on "open assessment" for the first correction round, when the "open assessment" button is also visible in the second column. This should open the first result correctly.
3. 

### Test Coverage




### Screenshots
